### ISSUE TITLE
Supporting custom headers

### ISSUE THREAD

**Issue by JoshWillik** [funcool/buddy-sign#46]
I am developing a tool that works with Let's Encrypt's ACME server. I would like to use buddy for this, but `buddy.sign.jws/sign` does not support all the needed headers.

`jwk` and `kid` are easy enough since they're [in the RFC](https://tools.ietf.org/html/rfc7515#section-4.1.2), I could submit a PR for those.

I also need `nonce` and `url`, so how do you feel about modifying the API to allow arbitrary header values to be merged in? Would you accept a PR for this?

```
(buddy.sign.jws/sign "claims" "secret"
    {:header {:url "https://example.com" :nonce "some_id"}})
```

**Comment by niwinz** [comment]
Sorry for my late response. Yes, the PR will be welcome for that ;)

**Comment by niwinz** [comment]
Related to #44

**Comment by theikkila** [comment]
This is now implemented in #52
